[FIX] json 응답이 달라서 발생하는 문제

### 🐞 버그 요약

공공데이터에 더이상 값을 가지고 오지 못할때 생기는 문제

### 🔍 원인 분석

- dto는 내부클래스가 정의되어있는데 내부클래스의 값이 아무것도없어서 restTemplate에서 포매팅을 못하는 문제
- 기존 응답데이터는 다음과 같음
```
{
    "response": {
        "header": {
            "resultCode": "0000",
            "resultMsg": "OK"
        },
        "body": {
            "items": {
                "item": [
                    {
                        "contentId": "100000",
                        "serialnum": "441315",
                        "imageUrl": "https://gocamping.or.kr/upload/camp/100000/8961mOX5pkL8PIG296E1RU3S.jpg",
                        "createdtime": "2023-01-16 16:48:52",
                        "modifiedtime": "2023-01-16 16:48:52"
                    }
                ]
            },
            "numOfRows": 1,
            "pageNo": 1,
            "totalCount": 27
        }
    }
}
```

- 만약 데이터가 없다면 다음과 같이 응답이 옴
```
{
    "response": {
        "header": {
            "resultCode": "0000",
            "resultMsg": "OK"
        },
        "body": {
            "items": "",
            "numOfRows": 0,
            "pageNo": 22,
            "totalCount": 4072
        }
    }
}
```
- restTemplate을 이용하여 api를 호출할때 goCampingDataDto로 받게 되는데 데이터가 없다면 응답값이 바뀌기때문에 오류가 발생함

해결방안
1. Jackson 라이브러리를 이용한 CoercionConfig 설정
2. @JsonDeserialize 커스텀을 통한 구현
3. try-catch문을 통한 빈리스트 반환

그 중 가장 쉬운 3번을 통해서 해결함


```
[2024-12-04 05:29:47] [ERROR] [s.c.c.c.a.GlobalLoggingAspect.logAfterThrowing:39] - [예외 발생 메서드]: GoCampingService.getAndConvertToGoCampingDataDto(..) [예외]: Error while extracting response for type [class site.campingon.campingon.common.public_data.dto.GoCampingDataDto] and content type [application/json]
[2024-12-04 05:29:47] [ERROR] [s.c.c.c.a.GlobalLoggingAspect.logAfterThrowing:39] - [예외 발생 메서드]: GoCampingScheduler.scheduleCampDelete() [예외]: Error while extracting response for type [class site.campingon.campingon.common.public_data.dto.GoCampingDataDto] and content type [application/json]
[2024-12-04 05:29:47] [ERROR] [o.s.s.s.TaskUtils$LoggingErrorHandler.handleError:95] - Unexpected error occurred in scheduled task
org.springframework.web.client.RestClientException: Error while extracting response for type [class site.campingon.campingon.common.public_data.dto.GoCampingDataDto] and content type [application/json]

```

```
[2024-12-04 05:37:20] [ERROR] [s.c.c.c.a.GlobalLoggingAspect.logAfterThrowing:39] - [예외 발생 메서드]: GoCampingService.getAndConvertToGoCampingDataDto(..) [예외]: Error while extracting response for type [class site.campingon.campingon.common.public_data.dto.GoCampingDataDto] and content type [application/json]
[2024-12-04 05:37:20] [ERROR] [s.c.c.c.a.GlobalLoggingAspect.logAfterThrowing:39] - [예외 발생 메서드]: GoCampingController.createCampByGoCampingBasedList(..) [예외]: Error while extracting response for type [class site.campingon.campingon.common.public_data.dto.GoCampingDataDto] and content type [application/json]
[2024-12-04 05:37:20] [ERROR] [o.a.c.c.C.[.[.[.[dispatcherServlet].log:175] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: org.springframework.web.client.RestClientException: Error while extracting response for type [class site.campingon.campingon.common.public_data.dto.GoCampingDataDto] and content type [application/json]] with root cause
com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot coerce empty String ("") to `site.campingon.campingon.common.public_data.dto.GoCampingDataDto$Items` value (but could if coercion was enabled using `CoercionConfig`)
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 81] (through reference chain: site.campingon.campingon.common.public_data.dto.GoCampingDataDto["response"]->site.campingon.campingon.common.public_data.dto.GoCampingDataDto$Response["body"]->site.campingon.campingon.common.public_data.dto.GoCampingDataDto$Body["items"])
	
```

### ✔️ 수정 내용

_No response_

### 🔍 영향을 준 부분

_No response_

### 📌 참고 사항

_No response_

--- a/src/main/java/site/campingon/campingon/common/exception/ErrorCode.java
+++ b/src/main/java/site/campingon/campingon/common/exception/ErrorCode.java
@@ -57,7 +57,10 @@ public enum ErrorCode {
 
     FILE_NOT_FOUND(HttpStatus.NOT_FOUND, "FILE-005", "요청한 파일을 찾을 수 없습니다."),
     S3_OPERATION_FAILED(HttpStatus.INTERNAL_SERVER_ERROR, "FILE-006", "S3 작업 중 오류가 발생했습니다."),
-    FILE_MOVE_FAILED(HttpStatus.INTERNAL_SERVER_ERROR, "FILE-007", "파일 이동 중 오류가 발생했습니다.");
+    FILE_MOVE_FAILED(HttpStatus.INTERNAL_SERVER_ERROR, "FILE-007", "파일 이동 중 오류가 발생했습니다."),
+
+    GO_CAMPING_BAD_REQUEST(HttpStatus.BAD_REQUEST, "GOCAMPING-001", "잘못된 고캠핑 데이터 요청입니다."),
+    GO_CAMPING_IMAGE_BAD_REQUEST(HttpStatus.BAD_REQUEST, "GOCAMPING-002", "DB에 저장된 캠프가 없거나 잘못된 데이터 요청입니다.");
 
     private final HttpStatus httpStatus;
     private final String code;

--- a/src/main/java/site/campingon/campingon/common/public_data/controller/GoCampingController.java
+++ b/src/main/java/site/campingon/campingon/common/public_data/controller/GoCampingController.java
@@ -4,6 +4,7 @@ import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.*;
 import org.springframework.web.bind.annotation.*;
+import site.campingon.campingon.common.exception.GlobalException;
 import site.campingon.campingon.common.public_data.GoCampingPath;
 import site.campingon.campingon.common.public_data.dto.GoCampingDataDto;
 import site.campingon.campingon.common.public_data.dto.GoCampingImageDto;
@@ -13,6 +14,9 @@ import site.campingon.campingon.common.public_data.service.GoCampingService;
 
 import java.net.URISyntaxException;
 import java.util.List;
+
+import static site.campingon.campingon.common.exception.ErrorCode.GO_CAMPING_BAD_REQUEST;
+import static site.campingon.campingon.common.exception.ErrorCode.GO_CAMPING_IMAGE_BAD_REQUEST;
 
 /**
  * https://www.data.go.kr/tcs/dss/selectApiDataDetailView.do?publicDataPk=15101933
@@ -28,40 +32,51 @@ public class GoCampingController {
 
     /**
      * 고캠핑데이터 DB 저장
-     * */
+     */
     @PostMapping("/basedList")
     public ResponseEntity<List<GoCampingParsedResponseDto>> createCampByGoCampingBasedList(
             @RequestParam("numOfRows") Long numOfRows,  //한 페이지 결과 수
             @RequestParam("pageNo") Long pageNo)    //현재 페이지 번호
-            throws URISyntaxException {
-        //공공데이터를 조회하고 반환
-        GoCampingDataDto goCampingDataDto = goCampingService.getAndConvertToGoCampingDataDto(
-                GoCampingPath.BASED_LIST,
-                "numOfRows", numOfRows.toString(),
-                "pageNo", pageNo.toString());
+    {
+        try {
+            //공공데이터를 조회하고 반환
+            GoCampingDataDto goCampingDataDto = goCampingService.getAndConvertToGoCampingDataDto(
+                    GoCampingPath.BASED_LIST,
+                    "numOfRows", numOfRows.toString(),
+                    "pageNo", pageNo.toString());
 
             //Camp 관련 엔티티를 생성하고 DB에 저장한다.
             List<GoCampingParsedResponseDto> goCampingParsedResponseDtos
                     = goCampingService.createOrUpdateCampByGoCampingData(goCampingDataDto);
 
-        return ResponseEntity.status(HttpStatus.OK).body(goCampingParsedResponseDtos);
+            return ResponseEntity.status(HttpStatus.OK).body(goCampingParsedResponseDtos);
+        } catch (Exception e) { //
+            log.error("고캠핑데이터 저장 실패, 고캠핑 API 파라미터나 서비스키를 다시 확인해주세요");
+            throw new GlobalException(GO_CAMPING_BAD_REQUEST);
+        }
     }
 
     /**
      * DB에 Camp Id를 가져와서 Id가 가지고있는 이미지를 호출 및 저장
-     * */
+     */
     @PostMapping("/imageList")
     public ResponseEntity<List<List<GoCampingImageParsedResponseDto>>> createCampImageByGoCampingImageList(
             @RequestParam("imageCnt") long imageCnt)    //몇개의 이미지개수를 갖고올지
-            throws URISyntaxException {
-        //공공데이터를 조회하고 dto로 변환
-        List<GoCampingImageDto> goCampingImageDto = goCampingService.getAndConvertToAllGoCampingImageDataDto(imageCnt);
+    {
+        try {
+            //공공데이터를 조회하고 dto로 변환
+            List<GoCampingImageDto> goCampingImageDto = goCampingService.getAndConvertToAllGoCampingImageDataDto(imageCnt);
 
-        //CampImage 를 생성하고 DB에 저장한다.
-        List<List<GoCampingImageParsedResponseDto>> goCampingParsedResponseDtos
-                = goCampingService.createOrUpdateCampImageByGoCampingImageData(goCampingImageDto);
+            //CampImage 를 생성하고 DB에 저장한다.
+            List<List<GoCampingImageParsedResponseDto>> goCampingParsedResponseDtos
+                    = goCampingService.createOrUpdateCampImageByGoCampingImageData(goCampingImageDto);
 
-        return ResponseEntity.status(HttpStatus.OK).body(goCampingParsedResponseDtos);
+            return ResponseEntity.status(HttpStatus.OK).body(goCampingParsedResponseDtos);
+        } catch (Exception e) { //
+            log.error("고캠핑데이터 저장 실패, 고캠핑 API 파라미터나 서비스키를 다시 확인해주세요");
+            log.error("DB에 캠프데이터가 있는지 확인해주세요");
+            throw new GlobalException(GO_CAMPING_IMAGE_BAD_REQUEST);
+        }
     }
 
 //    //위치기반정보 목록 조회

--- a/src/main/java/site/campingon/campingon/common/public_data/controller/GoCampingController.java
+++ b/src/main/java/site/campingon/campingon/common/public_data/controller/GoCampingController.java
@@ -55,7 +55,7 @@ public class GoCampingController {
             @RequestParam("imageCnt") Long imageCnt)    //몇개의 이미지개수를 갖고올지
             throws URISyntaxException {
         //공공데이터를 조회하고 dto로 변환
-        List<GoCampingImageDto> goCampingImageDto = goCampingService.getAndConvertToGoCampingImageDataDto(imageCnt);
+        List<GoCampingImageDto> goCampingImageDto = goCampingService.getAndConvertToAllGoCampingImageDataDto(imageCnt);
 
         //CampImage 를 생성하고 DB에 저장한다.
         List<List<GoCampingImageParsedResponseDto>> goCampingParsedResponseDtos

--- a/src/main/java/site/campingon/campingon/common/public_data/controller/GoCampingController.java
+++ b/src/main/java/site/campingon/campingon/common/public_data/controller/GoCampingController.java
@@ -52,7 +52,7 @@ public class GoCampingController {
      * */
     @PostMapping("/imageList")
     public ResponseEntity<List<List<GoCampingImageParsedResponseDto>>> createCampImageByGoCampingImageList(
-            @RequestParam("imageCnt") Long imageCnt)    //몇개의 이미지개수를 갖고올지
+            @RequestParam("imageCnt") long imageCnt)    //몇개의 이미지개수를 갖고올지
             throws URISyntaxException {
         //공공데이터를 조회하고 dto로 변환
         List<GoCampingImageDto> goCampingImageDto = goCampingService.getAndConvertToAllGoCampingImageDataDto(imageCnt);

--- a/src/main/java/site/campingon/campingon/common/public_data/dto/GoCampingDataDto.java
+++ b/src/main/java/site/campingon/campingon/common/public_data/dto/GoCampingDataDto.java
@@ -32,7 +32,8 @@ public class GoCampingDataDto {
     @JsonNaming(PropertyNamingStrategies.LowerCamelCaseStrategy.class)
     public static class Body {
         private Items items;
-        private Long numOfRows;
+        private long numOfRows;
+        private long totalCount;
     }
 
     @Getter

--- a/src/main/java/site/campingon/campingon/common/public_data/schedule/GoCampingScheduler.java
+++ b/src/main/java/site/campingon/campingon/common/public_data/schedule/GoCampingScheduler.java
@@ -23,8 +23,9 @@ import static site.campingon.campingon.common.public_data.PublicDataConstants.*;
  * 2일 : 데이터 업데이트
  * 3일 : 데이터 삭제
  * 스케줄러 테스트해보고싶다면 다음 어노테이션 사용하기
+ *
  * @Scheduled(initialDelay = 10000)    //스케줄러에 등록되고 10초뒤에 메소드 실행 -> 최초에 한번 실행
- * */
+ */
 
 @RequiredArgsConstructor
 @Component
@@ -32,24 +33,29 @@ import static site.campingon.campingon.common.public_data.PublicDataConstants.*;
 public class GoCampingScheduler {
     private final GoCampingService goCampingService;
 
-    private static final Long NUM_OF_ROWS = 500L;   //while 문이 한번돌때 저장되는 개수
-    private static final Long IMAGE_CNT = 10L;  //Camp id 하나에 저장될 이미지 개수
+    private static final long NUM_OF_ROWS = 500L;   //while 문이 한번돌때 저장되는 개수
+    private static final long IMAGE_CNT = 10L;  //Camp id 하나에 저장될 이미지 개수
+    //todo totalCnt 불러와서 처리하기, 컨트롤러에서는 Exception 처리 ( 혹은 서비스에서 처리)
 
     //신규 데이터 저장
     @Scheduled(cron = "0 0 1 * * ?", zone = "Asia/Seoul")  //매달 1일 오전 00:00에 실행
     public void scheduleCampCreation() {
         try {
-            Long pageNo = 1L;      // 현재 페이지 번호
+            long pageNo = 1L;      // 현재 페이지 번호
+            long totalCount = 0L;   //전체 데이터 개수
 
             log.info("캠프 생성 스케줄러 실행");
             while (true) {
                 //고캠핑 API 호출하고 dto 로 변환(이미지)
                 GoCampingDataDto goCampingDataDto = goCampingService.getAndConvertToGoCampingDataDto(
                         GoCampingPath.BASED_SYNC_LIST,
-                        "numOfRows", NUM_OF_ROWS.toString(),
-                        "pageNo", pageNo.toString(),
+                        "numOfRows", Long.toString(NUM_OF_ROWS),
+                        "pageNo", Long.toString(pageNo),
                         "syncStatus", SYNC_STATUS_NEW
                 );
+
+                //공공 API 가져올 총 cnt 추출
+                totalCount = goCampingDataDto.getResponse().getBody().getTotalCount();
 
                 List<GoCampingParsedResponseDto> goCampingParsedResponseDtos =
                         goCampingService.createOrUpdateCampByGoCampingData(goCampingDataDto);
@@ -71,12 +77,13 @@ public class GoCampingScheduler {
 
                 log.info("캠프 이미지 데이터 신규 생성 성공: " + goCampingImageParsedResponseDtos.size() + "개");
 
-                pageNo++;   //페이지 번호 증가
-
-                //더이상 가져올 데이터(Row)가 없다면 break
-                if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0) {
+                //더이상 가져올 데이터(Row)가 없거나 페이지번호 * 현재페이지수가 totalCnt 보다 크면 break
+                if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0 ||
+                        NUM_OF_ROWS * pageNo >= totalCount) {
                     break;
                 }
+
+                pageNo++;   //페이지 번호 증가
             }
         } catch (URISyntaxException e) {
             log.error("캠프 데이터 생성 실패: " + e.getMessage());
@@ -86,7 +93,8 @@ public class GoCampingScheduler {
     //데이터 업데이트
     @Scheduled(cron = "0 0 0 2 * ?", zone = "Asia/Seoul")   //매월 2일 00:00 실행
     public void scheduleCampUpdate() {
-        Long pageNo = 1L;      // 현재 페이지 번호
+        long pageNo = 1L;      // 현재 페이지 번호
+        long totalCount = 0L;   //전체 데이터 개수
 
         log.info("캠프 업데이트 스케줄러 실행");
 
@@ -94,10 +102,12 @@ public class GoCampingScheduler {
             while (true) {
                 GoCampingDataDto goCampingDataDto = goCampingService.getAndConvertToGoCampingDataDto(
                         GoCampingPath.BASED_SYNC_LIST,
-                        "numOfRows", NUM_OF_ROWS.toString(),
-                        "pageNo", pageNo.toString(),
+                        "numOfRows", Long.toString(NUM_OF_ROWS),
+                        "pageNo", Long.toString(pageNo),
                         "syncStatus", SYNC_STATUS_UPDATE
                 );
+
+                totalCount = goCampingDataDto.getResponse().getBody().getTotalCount();
 
                 List<GoCampingParsedResponseDto> goCampingParsedResponseDtos =
                         goCampingService.createOrUpdateCampByGoCampingData(goCampingDataDto);
@@ -116,11 +126,12 @@ public class GoCampingScheduler {
 
                 log.info("캠프 이미지 데이터 업데이트 성공: " + goCampingImageParsedResponseDtos.size() + "개");
 
-                pageNo++;   //페이지 번호 증가
-
-                if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0) {
+                if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0 ||
+                        NUM_OF_ROWS * pageNo >= totalCount) {
                     break;
                 }
+
+                pageNo++;   //페이지 번호 증가
             }
         } catch (URISyntaxException e) {
             log.error("캠프 데이터 업데이트 실패: " + e.getMessage());
@@ -131,27 +142,32 @@ public class GoCampingScheduler {
     //불필요한 데이터 삭제
     @Scheduled(cron = "0 0 0 3 * ?", zone = "Asia/Seoul")   //매월 3일 00:00 실행
     public void scheduleCampDelete() {
-        Long pageNo = 1L;      // 현재 페이지 번호
+        long pageNo = 1L;      // 현재 페이지 번호
+        long totalCount = 0L;   //전체 데이터 개수
 
         log.info("캠프 삭제 스케줄러 실행");
         try {
             while (true) {
                 GoCampingDataDto goCampingDataDto = goCampingService.getAndConvertToGoCampingDataDto(
                         GoCampingPath.BASED_SYNC_LIST,
-                        "numOfRows", NUM_OF_ROWS.toString(),
-                        "pageNo", pageNo.toString(),
+                        "numOfRows", Long.toString(NUM_OF_ROWS),
+                        "pageNo", Long.toString(pageNo),
                         "syncStatus", SYNC_STATUS_DELETE
                 );
 
+                totalCount = goCampingDataDto.getResponse().getBody().getTotalCount();
                 int deleteCnt = goCampingService.deleteCampByGoCampingData(goCampingDataDto);
 
                 log.info("캠프 데이터 삭제 성공: " + deleteCnt + "개");
 
                 pageNo++;   //페이지 번호 증가
 
-                if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0) {
+                if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0 ||
+                        NUM_OF_ROWS * pageNo >= totalCount) {
                     break;
                 }
+
+                pageNo++;   //페이지 번호 증가
             }
         } catch (URISyntaxException e) {
             log.error("캠프 데이터 삭제 실패: " + e.getMessage());

--- a/src/main/java/site/campingon/campingon/common/public_data/schedule/GoCampingScheduler.java
+++ b/src/main/java/site/campingon/campingon/common/public_data/schedule/GoCampingScheduler.java
@@ -32,7 +32,7 @@ import static site.campingon.campingon.common.public_data.PublicDataConstants.*;
 public class GoCampingScheduler {
     private final GoCampingService goCampingService;
 
-    private static final Long NUM_OF_ROWS = 10L;   //while 문이 한번돌때 저장되는 개수
+    private static final Long NUM_OF_ROWS = 500L;   //while 문이 한번돌때 저장되는 개수
     private static final Long IMAGE_CNT = 10L;  //Camp id 하나에 저장될 이미지 개수
 
     //신규 데이터 저장
@@ -56,13 +56,14 @@ public class GoCampingScheduler {
 
                 log.info("캠프 데이터 신규 생성 성공: " + goCampingParsedResponseDtos.size() + "개");
 
-
-                //todo 몇초 기다리기
-//                Thread.sleep(10000);  //일단보류
+                //캠프 아이디 추출
+                List<Long> campIdList = goCampingParsedResponseDtos.stream()
+                        .map(GoCampingParsedResponseDto::getContentId)
+                        .toList();
 
                 //고캠핑 API 호출하고 dto 로 변환(이미지)
                 List<GoCampingImageDto> goCampingImageDto
-                        = goCampingService.getAndConvertToGoCampingImageDataDto(IMAGE_CNT);
+                        = goCampingService.getAndConvertToGoCampingImageDataDto(campIdList, IMAGE_CNT);
 
                 //CampImage 를 생성하고 DB에 저장한다.
                 List<List<GoCampingImageParsedResponseDto>> goCampingImageParsedResponseDtos
@@ -76,8 +77,6 @@ public class GoCampingScheduler {
                 if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0) {
                     break;
                 }
-                //todo 기다리기 처리
-//                Thread.sleep(10000); //일단보류
             }
         } catch (URISyntaxException e) {
             log.error("캠프 데이터 생성 실패: " + e.getMessage());
@@ -103,9 +102,14 @@ public class GoCampingScheduler {
                 List<GoCampingParsedResponseDto> goCampingParsedResponseDtos =
                         goCampingService.createOrUpdateCampByGoCampingData(goCampingDataDto);
 
+                List<Long> campIdList = goCampingParsedResponseDtos.stream()
+                        .map(GoCampingParsedResponseDto::getContentId)
+                        .toList();
+
                 log.info("캠프 데이터 업데이트 성공: " + goCampingParsedResponseDtos.size() + "개");
 
-                List<GoCampingImageDto> goCampingImageDto = goCampingService.getAndConvertToGoCampingImageDataDto(IMAGE_CNT);
+                List<GoCampingImageDto> goCampingImageDto
+                        = goCampingService.getAndConvertToGoCampingImageDataDto(campIdList, IMAGE_CNT);
 
                 List<List<GoCampingImageParsedResponseDto>> goCampingImageParsedResponseDtos
                         = goCampingService.createOrUpdateCampImageByGoCampingImageData(goCampingImageDto);
@@ -117,8 +121,6 @@ public class GoCampingScheduler {
                 if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0) {
                     break;
                 }
-                //todo 몇초 기다리기
-//                Thread.sleep(10000);
             }
         } catch (URISyntaxException e) {
             log.error("캠프 데이터 업데이트 실패: " + e.getMessage());
@@ -150,8 +152,6 @@ public class GoCampingScheduler {
                 if (goCampingDataDto.getResponse().getBody().getNumOfRows() == 0) {
                     break;
                 }
-                //todo 몇초 기다리기
-//                Thread.sleep(10000);
             }
         } catch (URISyntaxException e) {
             log.error("캠프 데이터 삭제 실패: " + e.getMessage());

--- a/src/main/java/site/campingon/campingon/common/public_data/service/GoCampingService.java
+++ b/src/main/java/site/campingon/campingon/common/public_data/service/GoCampingService.java
@@ -184,7 +184,7 @@ public class GoCampingService {
 
     //공공데이터 이미지 API 조회하고 dto 변환(DB에 있는 모든 Camp 테이블의 이미지 조회)
     public List<GoCampingImageDto> getAndConvertToAllGoCampingImageDataDto(
-            Long imageCnt)
+            long imageCnt)
             throws URISyntaxException {
         List<GoCampingImageDto> goCampingDataDtoList = new ArrayList<>();
 
@@ -194,7 +194,7 @@ public class GoCampingService {
 
         for (Long campId : campIdList) {
             URI uri = goCampingProviderService.createUri(GoCampingPath.IMAGE_LIST,
-                    "numOfRows", imageCnt.toString(),
+                    "numOfRows", Long.toString(imageCnt),
                     "pageNo", IMAGE_PAGE_NO,  //몇번부터 시작할지
                     "contentId", campId.toString());
 
@@ -206,14 +206,14 @@ public class GoCampingService {
 
     //공공데이터 이미지 API 조회 및 Dto 변환(campIdList 에 해당하는 id만 이미지 조회)
     public List<GoCampingImageDto> getAndConvertToGoCampingImageDataDto(
-            List<Long> campIdList, Long imageCnt)
+            List<Long> campIdList, long imageCnt)
             throws URISyntaxException {
         List<GoCampingImageDto> goCampingDataDtoList = new ArrayList<>();
 
 
         for (Long campId : campIdList) {
             URI uri = goCampingProviderService.createUri(GoCampingPath.IMAGE_LIST,
-                    "numOfRows", imageCnt.toString(),
+                    "numOfRows", Long.toString(imageCnt),
                     "pageNo", IMAGE_PAGE_NO,  //몇번부터 시작할지
                     "contentId", campId.toString());
 

--- a/src/main/java/site/campingon/campingon/common/public_data/service/GoCampingService.java
+++ b/src/main/java/site/campingon/campingon/common/public_data/service/GoCampingService.java
@@ -182,8 +182,8 @@ public class GoCampingService {
         return restTemplate.getForObject(uri, GoCampingDataDto.class);  //API 호출
     }
 
-    //공공데이터 이미지 API 조회하고 dto 변환
-    public List<GoCampingImageDto> getAndConvertToGoCampingImageDataDto(
+    //공공데이터 이미지 API 조회하고 dto 변환(DB에 있는 모든 Camp 테이블의 이미지 조회)
+    public List<GoCampingImageDto> getAndConvertToAllGoCampingImageDataDto(
             Long imageCnt)
             throws URISyntaxException {
         List<GoCampingImageDto> goCampingDataDtoList = new ArrayList<>();
@@ -191,6 +191,25 @@ public class GoCampingService {
         List<Long> campIdList = campRepository.findAll().stream()
                 .map(Camp::getId)
                 .toList();
+
+        for (Long campId : campIdList) {
+            URI uri = goCampingProviderService.createUri(GoCampingPath.IMAGE_LIST,
+                    "numOfRows", imageCnt.toString(),
+                    "pageNo", IMAGE_PAGE_NO,  //몇번부터 시작할지
+                    "contentId", campId.toString());
+
+            goCampingDataDtoList.add(
+                    restTemplate.getForObject(uri, GoCampingImageDto.class)); //API 호출
+        }
+        return goCampingDataDtoList;
+    }
+
+    //공공데이터 이미지 API 조회 및 Dto 변환(campIdList 에 해당하는 id만 이미지 조회)
+    public List<GoCampingImageDto> getAndConvertToGoCampingImageDataDto(
+            List<Long> campIdList, Long imageCnt)
+            throws URISyntaxException {
+        List<GoCampingImageDto> goCampingDataDtoList = new ArrayList<>();
+
 
         for (Long campId : campIdList) {
             URI uri = goCampingProviderService.createUri(GoCampingPath.IMAGE_LIST,

--- a/src/main/java/site/campingon/campingon/common/util/AuthenticateUser.java
+++ b/src/main/java/site/campingon/campingon/common/util/AuthenticateUser.java
@@ -5,10 +5,14 @@ import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Component;
 import site.campingon.campingon.common.jwt.CustomUserDetails;
 
+/**
+ * SecurityContextHolder 직접 다루기
+ * */
+
 @Component
 public class AuthenticateUser {
 
-
+    //인증된 사용자면 userId, 그렇지않으면 0 반환
     public Long authenticateUserId() {
         Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
 

